packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -83,869 +109,812 @@ if (__DEV__) {
   didWarnAboutStatelessRefs = {};
 }
 
-export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
-  config: HostConfig<T, P, I, TI, HI, PI, C, CC, CX, PL>,
-  hostContext: HostContext<C, CX>,
-  legacyContext: LegacyContext,
-  newContext: NewContext,
-  hydrationContext: HydrationContext<C, CX>,
-  scheduleWork: (fiber: Fiber, expirationTime: ExpirationTime) => void,
-  computeExpirationForFiber: (
-    startTime: ExpirationTime,
-    fiber: Fiber,
-  ) => ExpirationTime,
-  profilerTimer: ProfilerTimer,
-  recalculateCurrentTime: () => ExpirationTime,
-) {
-  const {shouldSetTextContent, shouldDeprioritizeSubtree} = config;
-
-  const {pushHostContext, pushHostContainer} = hostContext;
-
-  const {
-    pushProvider,
-    getContextCurrentValue,
-    getContextChangedBits,
-  } = newContext;
-
-  const {
-    markActualRenderTimeStarted,
-    stopBaseRenderTimerIfRunning,
-  } = profilerTimer;
-
-  const {
-    getMaskedContext,
-    getUnmaskedContext,
-    hasContextChanged: hasLegacyContextChanged,
-    pushContextProvider: pushLegacyContextProvider,
-    pushTopLevelContextObject,
-    invalidateContextProvider,
-  } = legacyContext;
-
-  const {
-    enterHydrationState,
-    resetHydrationState,
-    tryToClaimNextHydratableInstance,
-  } = hydrationContext;
-
-  const {
-    adoptClassInstance,
-    constructClassInstance,
-    mountClassInstance,
-    resumeMountClassInstance,
-    updateClassInstance,
-  } = ReactFiberClassComponent(
-    legacyContext,
-    scheduleWork,
-    computeExpirationForFiber,
-    memoizeProps,
-    memoizeState,
-    recalculateCurrentTime,
+// TODO: Remove this and use reconcileChildrenAtExpirationTime directly.
```

Comment:
Any more context for this TODO? (What's the blocker?)

Reply:
No idea. I didn't add this, the code just moved from below.